File: SIH/cyber-risk-platform/backend/app/schemas/risk.py
from datetime import datetime
from uuid import UUID
from typing import Any, Optional
from pydantic import BaseModel, Field

class RiskFactors(BaseModel):
    impact: Optional[float] = None
    likelihood: Optional[float] = None
    gross_risk: Optional[float] = None
    mitigation_factor: Optional[float] = None
    total_assets: Optional[int] = None
    critical_assets: Optional[int] = None

class RiskMetadata(BaseModel):
    factors: RiskFactors = Field(default_factory=RiskFactors)
    drivers: list[str] = Field(default_factory=list)
    explanation: str = ""
    confidence: int = Field(0, ge=0, le=100)

class RiskScoreBase(BaseModel):
    score: float = Field(..., ge=0, le=100)
    risk_level: str
    calculation_version: Optional[str] = None
    risk_metadata: Optional[RiskMetadata] = None

class RiskScoreCreate(RiskScoreBase):
    organization_id: UUID
    asset_id: Optional[UUID] = None

class RiskScoreRead(RiskScoreBase):
    id: UUID
    organization_id: UUID
    asset_id: Optional[UUID] = None
    calculated_at: datetime
    created_at: datetime

    model_config = {
        "from_attributes": True,
        "populate_by_name": True
    }

class RiskTrendDataPoint(BaseModel):
    timestamp: datetime
    score: float
    risk_level: str

class RiskTrendResponse(BaseModel):
    current_score: RiskScoreRead
    historical_trend: list[RiskTrendDataPoint]

File: SIH/cyber-risk-platform/frontend/src/types/risk.ts
export interface RiskFactors {
  impact: number;
  likelihood: number;
  gross_risk: number;
  mitigation_factor: number;
  total_assets?: number;
  critical_assets?: number;
}

export interface RiskMetadata {
  factors: RiskFactors;
  drivers: string[];
  explanation: string;
  confidence: number;
}

export interface RiskScore {
  id: string;
  organization_id: string;
  asset_id?: string;
  score: number;
  risk_level: string;
  calculation_version: string;
  metadata_: RiskMetadata;
  calculated_at: string;
  created_at: string;
}

export interface RiskTrendDataPoint {
  timestamp: string;
  score: number;
  risk_level: string;
}

export interface RiskTrendResponse {
  current_score: RiskScore;
  historical_trend: RiskTrendDataPoint[];
}
